pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java

```diff
@@ -0,0 +1,408 @@
+
+    private final String ddpParticipantId;
+
+    @ColumnName(DBConstants.PARTICIPANT_ID)
+    private final Long participantId;
+
+    @ColumnName(DBConstants.FILE_NAME)
+    private final String fileName;
+
+    @ColumnName(DBConstants.MIME_TYPE)
+    private final String mimeType;
+
+    private final String bucket;
+
+    private final String blobPath;
+
+    @ColumnName(DBConstants.CREATED_BY_USER_ID)
+    private final Long createdByUserId;
+
+    @ColumnName(DBConstants.CREATED_AT)
+    private final long createdAt;
+
+    @ColumnName(DBConstants.DELETED_BY_USER_ID)
+    private final Long deletedByUserId;
+
+    @ColumnName(DBConstants.DELETED_AT)
+    private final long deletedAt;
+
+    @ColumnName(DBConstants.IS_VIRUS_FREE)
+    private Boolean isVirusFree;
+
+    private static final String SQL_SELECT_DOCUMENTS_BASE = "SELECT sD.id, p.ddp_instance_id, "
+            + "p.ddp_participant_id, p.participant_id, sD.file_name, sD.mime_type, sD.bucket, sD.blob_path, "
+            + "sD.created_by_user_id, sD.created_at, sD.deleted_by_user_id, sD.deleted_at, sD.is_virus_free "
+            + "FROM  somatic_documents sD "
+            + "LEFT JOIN ddp_instance as ddp on (ddp.ddp_instance_id = sD.ddp_instance_id) "
+            + "LEFT JOIN ddp_participant p on (p.participant_id = sD.participant_id) ";
+
+    private static final String SQL_SELECT_DOCUMENTS_BY_REALM = SQL_SELECT_DOCUMENTS_BASE
+            + "WHERE ddp.instance_name = ?";
+
+    private static final String SQL_SELECT_DOCUMENTS_BY_REALM_AND_PTPT = SQL_SELECT_DOCUMENTS_BASE
+            + "WHERE ddp.instance_name = ? AND ddp_participant_id = ?";
+
+    private static final String SQL_SELECT_DOCUMENT_BY_ID = SQL_SELECT_DOCUMENTS_BASE
+            + "WHERE sD.id = ?";
+
+    private static final String SQL_SELECT_DOCUMENT_BY_BUCKET_AND_PATH = SQL_SELECT_DOCUMENTS_BASE
+            + "WHERE sD.bucket = ? AND sD.blob_path = ?";
+
+    private static final String SQL_UPDATE_VIRUS_STATUS_SUCCESS = "UPDATE somatic_documents SET is_virus_free = ?, "
+            + "bucket = ?, blob_path = ? WHERE bucket = ? AND blob_path = ?";
+
+    private static final String SQL_UPDATE_VIRUS_STATUS_FAILED = "UPDATE somatic_documents SET is_virus_free = ?, "
+            + "deleted_at = ? WHERE bucket = ? AND blob_path = ?";
+
+    private static final String SQL_DELETE_DOCUMENT_BY_DOCUMENT_ID = "UPDATE somatic_documents SET deleted_by_user_id = ?, "
+            + "deleted_at = ? WHERE id = ?";
+
+    private static final String SQL_INSERT_SOMATIC_DOCUMENT = "INSERT INTO somatic_documents SET ddp_instance_id = "
+            + "(SELECT ddp_instance_id FROM ddp_instance WHERE instance_name = ?), file_name = ?, "
+            + "mime_type = ?, bucket = ?, blob_path = ?, created_by_user_id = ?, created_at= ?, "
+            + "participant_id = (SELECT p.participant_id from ddp_participant p "
+            + "LEFT JOIN ddp_instance as ddp ON (ddp.ddp_instance_id = p.ddp_instance_id) "
```

Comment:
What happens if a user deletes a file, then creates a new version of the file with the same name?

Reply:
It saves as expected.  There's a UUID generated on the blobPath but no constraint that the "fileName" has to be unique.

Reply:
I see that we don't currently offer 'by filename' query criteria, so this seems fine.
Thx for the explanation.